src/storage/store/rel_table_data.cpp

```diff
@@ -413,22 +447,24 @@ void RelTableData::prepareCommitForCSRColumns(
             // Thus, we can fall back to directly update the adj column and property columns based
             // on csr offsets.
             prepareCommitCSRNGWithoutSliding(transaction, nodeGroupIdx, relNodeGroupInfo,
-                csrOffsetChunk.get(), relIDChunk.get(), relNG);
+                csrOffsetChunk.get(), csrLengthChunk.get(), relIDChunk.get(), relNG);
         } else {
             // We need to update the csr offset column. Thus, we cannot simply fall back to directly
             // update the adj column and property columns based on csr offsets.
             prepareCommitCSRNGWithSliding(transaction, nodeGroupIdx, relNodeGroupInfo,
-                csrOffsetChunk.get(), relIDChunk.get(), relNG);
+                csrOffsetChunk.get(), csrLengthChunk.get(), relIDChunk.get(), relNG);
         }
     }
 }
 
-static std::pair<offset_t, offset_t> getCSRStartAndEndOffset(
-    offset_t offsetInNodeGroup, const ColumnChunk& csrOffsets) {
-    return offsetInNodeGroup == 0 ?
-               std::make_pair((offset_t)0, csrOffsets.getValue<offset_t>(offsetInNodeGroup)) :
-               std::make_pair(csrOffsets.getValue<offset_t>(offsetInNodeGroup - 1),
-                   csrOffsets.getValue<offset_t>(offsetInNodeGroup));
+static std::pair<offset_t, offset_t> getCSRStartAndEndOffset(offset_t offsetInNodeGroup,
+    const ColumnChunk& csrOffsetChunk, const ColumnChunk& csrLengthChunk) {
+    auto endOffset = csrOffsetChunk.getValue<offset_t>(offsetInNodeGroup);
+    auto length = csrLengthChunk.getValue<offset_t>(offsetInNodeGroup);
```

Comment:
Using getValue<offset_t> may cause some confusion there. Should we introduce a new alias length_t?

Reply:
Sounds good to me.